FluidNC/src/Spindles/Spindle.cpp

```diff
@@ -108,6 +120,43 @@ namespace Spindles {
         _speeds.push_back({ max, 100.0f });
     }
 
+    const char* Spindle::atc_info() {  // this can be used in the startup response
+        std::string atc_info = "";
+        if (_atc != NULL) {
+            atc_info = " ATC:" + _atc_name;
+        } else if (!_m6_macro._gcode.empty()) {
+            atc_info = " m6_macro:";
+        }
+        return atc_info.c_str();
```

Comment:
This isn't safe.  The atc_info string object goes out of scope and is freed when atc_info() returns, and the c_str() pointer is then pointing to memory that is now in the heap and could be reused at any time.  You could make it `std::string Spindle::atc_info()`